nemo/collections/asr/models/ctc_models.py

```diff
@@ -78,8 +77,8 @@ def setup_test_data(self, test_data_layer_params: Optional[Dict]):
             test_data_layer_params['shuffle'] = False
         self.__test_dl = self.__setup_dataloader_from_config(config=test_data_layer_params)
 
-    def setup_optimization(self, optim_params: Optional[Dict]):
-        self.__optimizer = torch.optim.Adam(self.parameters(), lr=optim_params['lr'])
+    def setup_optimization(self, optim_params: Optional[Dict] = None) -> torch.optim.Optimizer:
```

Comment:
Do we need to have setup_optimization here if it is written already in the higher clas?

Reply:
I added if because it's needed for a later PR of schedulers